src/components/EnvironmentOverride/EnvironmentOverride.tsx

```diff
@@ -1,69 +1,123 @@
-
-    const [deploymentState, setDeploymentState] = useState<ComponentStates>(null)
-    const [configMapState, setConfigMapState] = useState<ComponentStates>(null);
-    const [secretState, setSecretState] = useState<ComponentStates>(null);
+export default function EnvironmentOverride({ environmentsLoading, environments }: EnvironmentOverrideComponentProps) {
+    const params = useParams<{ appId: string; envId: string }>()
+    const [viewState, setViewState] = useState<ComponentStates>(null)
     const { path } = useRouteMatch()
     const { push } = useHistory()
+    const location = useLocation()
     const { environmentId, setEnvironmentId } = useAppContext()
+    const [headingData, setHeadingData] = useState<SectionHeadingType>()
 
-    const environments = mapByKey(appEnvironmentResult?.result || [], 'environmentId')
+    const environmentsMap = mapByKey(environments || [], 'environmentId')
 
     useEffect(() => {
         if (params.envId) setEnvironmentId(+params.envId)
-        setDeploymentState('loading');
-        setConfigMapState('loading');
-        setSecretState('loading');
+        setViewState('loading')
     }, [params.envId])
 
+    useEffect(() => {
+        if (location.pathname.includes(URLS.APP_CM_CONFIG)) {
+            setHeadingData(SECTION_HEADING_INFO[URLS.APP_CM_CONFIG])
+        } else if (location.pathname.includes(URLS.APP_CS_CONFIG)) {
+            setHeadingData(SECTION_HEADING_INFO[URLS.APP_CS_CONFIG])
+        } else {
+            setHeadingData(null)
+        }
+    }, [location.pathname])
+
     const envMissingRedirect = () => {
-        if (params.envId || appEnvironmentsLoading) return;
-        if (environments.has(environmentId)) {
-            const newUrl = generatePath(path, { appId: params.appId, envId: environmentId });
-            push(newUrl);
+        if (params.envId || environmentsLoading) return
+        if (environmentsMap.has(environmentId)) {
+            const newUrl = generatePath(path, { appId: params.appId, envId: environmentId })
+            push(newUrl)
         } else {
-            const workflowUrl = getAppComposeURL(params.appId, APP_COMPOSE_STAGE.WORKFLOW_EDITOR);
-            push(workflowUrl);
+            const workflowUrl = getAppComposeURL(params.appId, APP_COMPOSE_STAGE.WORKFLOW_EDITOR)
+            push(workflowUrl)
         }
     }
-    useEffect(envMissingRedirect, [appEnvironmentsLoading])
+    useEffect(envMissingRedirect, [environmentsLoading])
 
-    if (appEnvironmentsLoading) return <Progressing pageLoader />
-    if (deploymentState === 'failed' || configMapState === 'failed' || secretState === 'failed') {
-        return <Reload reload={event => { setConfigMapState('loading'); setSecretState('loading'); setDeploymentState('loading') }} />
+    if (!params.envId) {
+        return null
+    } else if (environmentsLoading) {
+        return <Progressing pageLoader />
+    } else if (viewState === 'failed') {
```

Comment:
Please make enum for all status or can reuse if exist

Reply:
The existing one was the type so converted it to an enum of a similar name - `ComponentStates`